src/main/java/org/zalando/nakadi/service/FeatureToggleServiceZk.java

```diff
@@ -1,57 +1,62 @@
-package org.zalando.nakadi.util;
+package org.zalando.nakadi.service;
 
-import com.google.common.cache.CacheBuilder;
-import com.google.common.cache.CacheLoader;
-import com.google.common.cache.LoadingCache;
 import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.cache.PathChildrenCache;
 import org.apache.zookeeper.KeeperException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.zalando.nakadi.repository.zookeeper.ZooKeeperHolder;
 
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.TimeUnit;
+import javax.annotation.PreDestroy;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
 
 public class FeatureToggleServiceZk implements FeatureToggleService {
 
     private static final Logger LOG = LoggerFactory.getLogger(FeatureToggleService.class);
-    private static final String PREFIX = "/nakadi/feature_toggle/";
+    private static final String PREFIX = "/nakadi/feature_toggle";
 
     private final ZooKeeperHolder zkHolder;
-    private final LoadingCache<Feature, Boolean> cache;
+    private PathChildrenCache featuresCache;
+    private final Map<Feature, String> featurePaths;
 
     public FeatureToggleServiceZk(final ZooKeeperHolder zkHolder) {
         this.zkHolder = zkHolder;
+        try {
+            this.featuresCache = new PathChildrenCache(zkHolder.get(), PREFIX, false);
+            this.featuresCache.start();
+        } catch (final Exception e) {
+            LOG.error(e.getMessage(), e);
+        }
+        featurePaths = new HashMap<>(Feature.values().length);
+        for (final Feature feature: Feature.values()) {
+            featurePaths.put(feature, PREFIX + "/" + feature.getId());
```

Comment:
What is the idea behind this map? To say the truth I don't understand why we need it.

Reply:
@adyach suggested that, since there are a small number of features, we build a map of paths at startup, in order to speed up lookups at runtime.

Reply:
@lmontrieux @adyach 
To say the truth I don't know what is faster to concat 3 strings or to make a hash and lookup element in a hashmap; for me both of these operations sound like nothing. But I know which code is simpler and doesn't confuse :) 
(BTW there is a helper method from Zookeeper ZKPaths.makePath(s1, s2) which uses StringBuilder inside)

Reply:
@v-stepanov suggestion was to not concatenate strings "here and there". Since this code path is heavily used, pre creating them before the usage should be beneficial. And I have no idea if JVM will optimise it for us. I do not see any complexity in Lionel's approach.

Reply:
@adyach Well, it will be beneficial only if it is faster. If it is slower - it will not :)
I don't say that current code is complex I say that it's more complicated than the one without this pre-created map.
I think it's a good example of premature optimization.

Reply:
It's good for me.